### ISSUE TITLE
[management-api] Unable to change admin password with the environment variables in Openshift

### ISSUE THREAD

**Issue by yoba730** [gravitee-io/issues#2680]
### Current Behavior

When I'm trying to change the admin password with env vars in openshift it doesn't work and it's keeps me the default password.

From what I understood it's seems to work fine with this format "gravitee.security.providers[0].users[1].password" but not with this one "gravitee_security_providers_0_users_1_password" and Openshift doesn't accept '.' and '[ ]'

### Possible Solution

if you can fix the problem of management-api to work with this format "gravitee_..._..." it would be perfect 

### Steps to Reproduce (for bugs)

1. add env vars gravitee_security_providers_0_password_encoding_algo in openshift deployments  to allow using an unencrypted password
2. add env vars gravitee_security_providers_0_users_1_password to only change the admin password

### Context

I'm trying to change the admin password of the management api using env vars in openshift 

### Your Environment

* Version used: 1.25.11

**Comment by tpenne** [comment]
I have exactly the same problem, on Openshift too. 
Password update seems to work only for the first user : 
- gravitee_security_providers_0_users_0_password declaration can be used to update "user" password - gravitee_security_providers_0_users_1_password declaration does not update "admin" password

**Version used** : 1.30.0

King Regards
Thomas

**Comment by aelamrani** [comment]
Hi,

what about something like this: `gravitee_security_providers[0].users[1].password`?

**Comment by tpenne** [comment]
Hi,
There are two different problems :

- The first one is that the way to write the environment variable provided in documentation (the second one, with this exact example : gravitee.security.providers[0].users[1].password), the only way that seems to work to update admin password according to multiple threads on the web, is not accepted by Openshift : 

> A valid environment variable name must consist of alphabetic characters, digits, '_', '-', or '.', and must not start with a digit.

Brackets [] are not allowed

- The second one is that the other way to write the environment variable : gravitee_security_providers_0_users_1_password seems to work only for the first user (with 0 instead of 1) and not for the subsequent one (password stays the same)

**Comment by tpenne** [comment]
Hi,
I performed multiple tests today and here are the results. I tested : 
- on Openshift
- with version 1.30.0
- using gravitee_security_providers_0_users_X_password syntax
- using different numbers and orders of users in gravitee.yml

Users in configuration | Password can be updated
------------------------ | ----------------------------
user | Y
admin | N
api1 | N
application1 | Y

Users in configuration | Password can be updated
------------------------ | ----------------------------
admin | N

Users in configuration | Password can be updated
------------------------ | ----------------------------
admin | N
user | Y

So it seems syntax works and not only for the first user. But it is not allowed to override users with ADMIN or API_PUBLISHER role.
